### PR TITLE
Fixes to osquery module

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
When running osquery commands through cmd.run we should pass python_shell=True to ensure everything is formatted right. 

### What issues does this PR fix or reference?
#42873

### Previous Behavior
Previous the osquery module would not work correctly on Windows beginning with the 2017.7 release, most likely related to the deprecation of the cmd.shell alias for cmd.run.

### New Behavior
This change forces python_shell=True to ensure everything is formatted correctly when passed to the cmd.run_all function.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/osquery.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _osquery(sql, format='json'):
     }
 
     cmd = 'osqueryi --json "{0}"'.format(sql)
-    res = __salt__['cmd.run_all'](cmd)
+    res = __salt__['cmd.run_all'](cmd, python_shell=True)
```
Comment (cachedout) [inline salt/modules/osquery.py]: We want to be careful about command injection here. We want to avoid somebody being able to pass in something that can close the double-quotes and then inject additional command-line arguments which are unrelated to osquery. Something like shlex might be a good tool here.

Comment (garethgreenaway) [inline salt/modules/osquery.py]: Good point.  Will find a better solution for this.  Thanks.

Comment (morganwillcock) [inline salt/modules/osquery.py]: I think this may be caused by the shlex.split in the cmd module. Just bypassing that for Windows might fix the issue; setting the posix option to false doesn't seem to account for quote marks and just executing the command as a single string should preserve them as-is.

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Comment by garethgreenaway** [comment]
Go Go Jenkins!
